libqtile/command/base.py

```diff
@@ -54,14 +56,85 @@ class CommandException(Exception):
     """Error raised while executing a command"""
 
 
-class CommandObject(metaclass=abc.ABCMeta):
+class _CommandMeta(abc.ABCMeta):
+    def __new__(cls, name, bases, dct):
+        instance = super().__new__(cls, name, bases, dct)
+
+        if hasattr(cls, "_commands"):
+            return instance
+
+        commands = {}
+        cmd_s = set()
+
+        for c in bases + (cls,):
```

Comment:
```suggestion
        for c in bases + (instance,):
```

Reply:
Also looks like `bases` is only the directly inherited class, doesn't seem to be a full MRO.